shim/src/scan.js

```diff
@@ -73,9 +73,11 @@ export function walkObjects(start, visitor) {
 
     enqueue(getPrototypeOf(obj), obj, '__proto__');
     const descs = getOwnPropertyDescriptors(obj);
-    ownKeys(descs).forEach(name => {
+    for (let name of ownKeys(descs)) {
+      // todo: all iteration needs uncurried forEach protection
```

Comment:
using `for..of` means you also need to protect against `Symbol.iterator` being poisoned, no?

Reply:
Probably ,yeah.  I think we decided that we should just capture an uncurried-this version of `Array.forEach` in `commons.js` and use that everywhere we do iteration.